lib/ethash/keccakf800.c

```diff
@@ -0,0 +1,70 @@
+/* ethash: C/C++ implementation of Ethash, the Ethereum Proof of Work algorithm.
+ * Copyright 2018 Pawel Bylica.
+ * Licensed under the Apache License, Version 2.0. See the LICENSE file.
+ */
+
+#include <stdint.h>
+
+ /* Implementation based on:
+	https://github.com/mjosaarinen/tiny_sha3/blob/master/sha3.c
+	converted from 64->32 bit words*/
+const uint32_t keccakf_rndc[24] = {
+	0x00000001, 0x00008082, 0x0000808a, 0x80008000, 0x0000808b, 0x80000001,
+	0x80008081, 0x00008009, 0x0000008a, 0x00000088, 0x80008009, 0x8000000a,
+	0x8000808b, 0x0000008b, 0x00008089, 0x00008003, 0x00008002, 0x00000080,
+	0x0000800a, 0x8000000a, 0x80008081, 0x00008080, 0x80000001, 0x80008008
+};
+
+#define ROTL(x,n,w) (((x) << (n)) | ((x) >> ((w) - (n))))
+#define ROTL32(x,n) ROTL(x,n,32)	/* 32 bits word */
+
+void keccak_f800_round(uint32_t st[25], const int r)
+{
+
+	const uint32_t keccakf_rotc[24] = {
```

Comment:
These value are obviously wrong, probably for Keccak-f[1600].

Reply:
which ones? keccakf_rndc[] is for 32bit keccak_800 not 1600

Reply:
Exactly. You can rol 32-bit integer by 36 bits. These values are probably taken from Keccak-f[1600] while they should be from Keccak-f[800] spec.